### ISSUE TITLE
Proposal Vector Tile Schema v3 

### ISSUE THREAD

**Issue by lukasmartinelli** [osm2vectortiles/osm2vectortiles#393]
### Vision

Break compatibility with Mapbox Streets V7 and only support modified Mapbox Open Styles (OSM Bright and Basic) and iterate from there. We don’t plan to stay compatible with Mapbox Streets in future schema changes.

### Schema

[Take a look at the diff between future OSM2VectorTiles v3 and Mapbox Streets v7](https://www.diffchecker.com/vzv2ueam).
Use the **[gl-style-schema](https://github.com/lukasmartinelli/gl-style-schema)** Python script to check for schema differences.

- Layers
  - Remove `#airport_label`
  - Remove `#motorway_junction`
  - Remove `#mountain_peak`
  - Remove `#rail_station_label`

- Fields
  - Remove `iso_3166_1` from `#admin` 
  - Remove `underground` in `#building``
  - Rename `house_num` to `#addr`
  - Remove `layer` from `#road`
  - Remove `shield`, `iso_3166_2` from `#road_label`
  - Remove `len` from `#road`
  - Remove `parent`, `type` from `#country_label`
  - Remove `placement` in `#marine_label`
  - Remove `area` field from `#state_label`, `#water_label`, `#marine_label`
  - Remove `ldir` from `#place_label`

### Conceptual

- Rename `class`, `type` concept to better OSM terms like `category`, `subcategory`

**Comment by pnorman** [comment]
Why the removal of the layers? Will those features be in a different layer?

**Comment by lukasmartinelli** [comment]
> Why the removal of the layers? Will those features be in a different layer?

tldr: This is more about restructuring the data like we had it before Streets v7 (in release last december)

We could either go broad with the layers (and add many specialized ones) or have only a few and people can then do the filtering of the style (approach by MB Streets).

For example `#airport_label` is still very important but fits into the `#poi_layer` just as well (in fact half a year back we had it like that). And `#rail_station_label` is too limiting, the default imposm3 mapping for example contains a `#transport_poi_label` which would contain all transport POI for special styling (i like that idea).

`#motorway_junction` is a special case. It was directly suggested in order because it is in Mapbox Streets and not in ours #359. That is really something that is not needed in OSM Bright and only there because of MB Streets. That's why remove it.

`#mountain_peak` could move back into POI or we start thinking about a landmark layer (there are other special landmarks not only mountain peaks). No longer so sure about this one.

These are layers that are not yet widely used in styles so we have the flexibility to restructure them.

**Comment by pnorman** [comment]
> For example #airport_label is still very important but fits into the #poi_layer just as well (in fact half a year back we had it like that). And #rail_station_label is too limiting, the default imposm3 mapping for example contains a #transport_poi_label which would contain all transport POI for special styling (i like that idea).

The diff doesn't show plans to add airports to the `#poi_layer`. Can you add these details so we can evaluate the changes?

> `#mountain_peak` could move back into POI or we start thinking about a landmark layer (there are other special landmarks not only mountain peaks). No longer so sure about this one.

fwiw, in ClearTables I have a landform_point table which has natural=peak [and similar tags](https://github.com/ClearTables/ClearTables/blob/4d7105213ae375106d98b01c14ef7eb0885752f5/landform.lua#L10-L16)

**Comment by lukasmartinelli** [comment]
> The diff doesn't show plans to add airports to the #poi_layer. Can you add these details so we can evaluate the changes?

Yes you are right. I just created a map style and `#railway_station_label`and `#mountain_peak_label`, `#airport_label` do make sense in their separate layers (although the could be more general, your `landform_point` suggestion is a very good idea and a layer like `#transport_poi` could be another idea).

For a real discussion what needs to be changed for actual improvement (not just artificial crippling of compatibility) we first need to look at the current schema.

The copyright claims we have from #387 are the following abstract terms.

- multiple possible data sources for different features
- the particular selection of elements to incorporate from those sources
- our tagging schemes and arrangement of the data elements
- label placement methods
- decisions about prominence
- use of features at different zoom levels
- iconography decisions
- ranking of features

However I struggle to apply these as copyrightable problems to the schema and layers below...
We always iterated with a visual map of OSM Bright based on MB Streets and improved the OSM2VectorTiles schema until the map OSM Bright based on OSM2VectorTiles looked good compared to the original OSM bright.

I would really like to have feedback in #387 **what layers or attributes are a problem**.

## Current Schema

The current schema look like this. The notation is invented and derived from CartoCSS (`#layer` and `[attribute]`. What I removed in the proposal above are layers and features that are not needed to display BSD OSM Bright style.

```css
#admin
  [admin_level]
  [disputed]
  [maritime]
#aeroway
  [type]
#airport_label
  [maki]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [ref]
  [scalerank]
#barrier_line
  [class]
#building
  [underground]
#country_label
  [code]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [scalerank]
#housenum_label
  [house_num]
#landuse
  [class]
  [type]
#landuse_overlay
  [class]
  [type]
#marine_label
  [labelrank]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [placement]
#mountain_peak_label
  [elevation_ft]
  [elevation_m]
  [maki]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
#place_label
  [capital]
  [ldir]
  [localrank]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [scalerank]
  [type]
#poi_label
  [localrank]
  [maki]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [ref]
  [scalerank]
  [type]
#rail_station_label
  [maki]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [network]
#road
  [class]
  [oneway]
  [structure]
  [type]
#road_label
  [class]
  [len]
  [localrank]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [ref]
  [reflen]
  [shield]
#state_label
  [abbr]
  [area]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
#water_label
  [area]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
#waterway
  [class]
  [type]
#waterway_label
  [class]
  [name]
  [name_de]
  [name_en]
  [name_es]
  [name_fr]
  [name_ru]
  [name_zh]
  [type]
```

**Comment by ImreSamu** [comment]
my proposal :

*  Add [wheelchair](http://wiki.openstreetmap.org/wiki/Key:wheelchair)  info to the POI-s or `#poi_label` 

proposed values:
* 'yes' / 'no' / 'limited' / 'designated'   : according to the  [osm wiki](http://wiki.openstreetmap.org/wiki/Key:wheelchair) 
*  ' '  :  ( empty )  :   no info ;  no osm wheelchair key
*   '.'  (dot)            :  bad/other info :   osm wheelchair key exists , but the value NOT IN [ 'yes' / 'no' / 'limited' / 'designated'  ]  so maybe other value or  tagging error!    This is  for  minimal  QA purpose , if somebody want to add link to edit this type of  POIs  then this will be so much  easier ! 
 ( see taginfo for the long tail values  : http://taginfo.openstreetmap.org/keys/wheelchair#values )
  
 
The '2 missing  values'   [ 'no info' and 'bad code'  ] is new concept in osm2vectortiles , but in the statistics it is very common, and in the future I would like to create some QA report ( charts, reports, based on osm2vectortiles v3 schema. ) 

So my proposal for the long term:    Moving the project to more general use case : 
* adding minimal humanitarian(wheelchair)  info 
* adding minimal QA info  ( if possible )

**Comment by stirringhalo** [comment]
Not sure if this is the place to ask, but where is the source for cartography for v3?

**Comment by stirringhalo** [comment]
Is https://github.com/ClearTables/osm-clear the base for all tm2source styling direction? I'm happy to move my 3D building stuff over there or to a fork of it if yes.

**Comment by bobwallis** [comment]
What's the rationale behind losing the ['layer=*'](https://wiki.openstreetmap.org/wiki/Key:layer) tag on roads? It really helps with designing styles that look good with complex junctions.
